experimental/packages/opentelemetry-sdk-metrics-base/src/export/PeriodicExportingMetricReader.ts

```diff
@@ -0,0 +1,109 @@
+import { ReaderResult, ReaderResultCode } from './ReaderResult';
+
+export type PeriodicExportingMetricReaderOptions = {
+  exporter: MetricExporter
+  exportIntervalMillis?: number,
+  exportTimeoutMillis?: number
+}
+
+/**
+ * {@link MetricReader} which collects metrics based on a user-configurable time interval, and passes the metrics to
+ * the configured {@link MetricExporter}
+ */
+export class PeriodicExportingMetricReader extends MetricReader {
+  private _interval?: ReturnType<typeof setInterval>;
+
+  private _exporter: MetricExporter;
+
+  private readonly _exportInterval: number;
+
+  private readonly _exportTimeout: number;
+
+  constructor(options: PeriodicExportingMetricReaderOptions) {
+    super(options.exporter.getPreferredAggregationTemporality());
+
+    if (options.exportIntervalMillis !== undefined && options.exportIntervalMillis <= 0) {
+      throw Error('exportIntervalMillis must be greater than 0');
+    }
+
+    if (options.exportTimeoutMillis !== undefined && options.exportTimeoutMillis <= 0) {
+      throw Error('exportTimeoutMillis must be greater than 0');
+    }
+
+    if (options.exportTimeoutMillis !== undefined &&
+      options.exportIntervalMillis !== undefined &&
+      options.exportIntervalMillis < options.exportTimeoutMillis) {
+      throw Error('exportIntervalMillis must be greater than or equal to exportTimeoutMillis');
+    }
+
+    this._exportInterval = options.exportIntervalMillis ?? 60000;
+    this._exportTimeout = options.exportTimeoutMillis ?? 30000;
+    this._exporter = options.exporter;
+  }
+
+  private async _runOnce(): Promise<void> {
+    const collectionResult = await new Promise<ReaderResult<MetricData[]>>(resolve =>
+      this.collect({ done: resolve }));
+
+    // Throw on any code that does not indicate success.
+    if (collectionResult.code !== ReaderResultCode.SUCCESS) {
+      throw collectionResult.error ?? new Error('Unknown error occurred during collection.');
+    }
+
+    // This case should never occur.
+    if (collectionResult.returnValue == null) {
+      throw new Error('Unknown error occurred during collection.');
+    }
+
+    await this._exporter.export(collectionResult.returnValue);
+  }
+
+  protected onInitialized(): void {
+    // start running the interval as soon as this reader is initialized and keep handle for shutdown.
+    this._interval = setInterval(async () => {
+      promiseWithTimeout(this._runOnce(), this._exportTimeout, (result => {
+        if (result.code === ReaderResultCode.TIMED_OUT) {
+          api.diag.error('Export took longer than %s milliseconds and timed out.', this._exportTimeout);
+          return;
+        }
+        if (result.code === ReaderResultCode.FAILED) {
+          api.diag.error('Unexpected error during export: %s', result.error);
+          return;
+        }
+      }));
+    }, this._exportInterval);
+  }
+
+  protected async onForceFlush(): Promise<void> {
+    await this._runOnce();
```

Comment:
IIUC this should delegate to exporter's forceFlush

Reply:
In the section of PushExporter.forceFlush [1], it is said that:

> This is a hint to ensure that the export of any Metrics the exporter has received prior to the call to ForceFlush SHOULD be completed as soon as possible


So to my understanding, the forceFlush should export the metric data received prior to the forceFlush call, but not a hint for a new collection.

[1]: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#forceflush-1